tests/clients/boto3/test_boto3_lambda.py

```diff
@@ -2,74 +2,256 @@
 # (c) Copyright Instana Inc. 2020
 
 from __future__ import absolute_import
+import unittest
+import json
 
-import os
 import boto3
-import pytest
-
 # TODO: Remove branching when we drop support for Python 3.7
-import sys
-if sys.version_info >= (3, 8):
+from sys import version_info
+if version_info >= (3, 8):
   from moto import mock_aws
 else:
-  from moto import mock_sqs as mock_aws
+  from moto import mock_lambda as mock_aws
 
-from instana.singletons import tracer
+from instana.singletons import tracer, agent
 from ...helpers import get_first_span_by_filter
 
+@unittest.skipIf(version_info < (3, 8), "Test skipped on Python < 3.8")
+class TestLambda(unittest.TestCase):
+    def setUp(self):
+        """ Clear all spans before a test run """
+        self.recorder = tracer.recorder
+        self.recorder.clear_spans()
+        self.mock = mock_aws(config={"lambda": {"use_docker": False}})
+        self.mock.start()
+        self.lambda_region = "us-east-1"
+        self.aws_lambda = boto3.client('lambda', region_name=self.lambda_region)
+        self.function_name = "myfunc"
+
+    def tearDown(self):
+        # Stop Moto after each test
+        self.mock.stop()
+
+
+    def test_lambda_invoke(self):
+        with tracer.start_active_span('test'):
+            result = self.aws_lambda.invoke(FunctionName=self.function_name, Payload=json.dumps({"message": "success"}))
+
+        self.assertEqual(result["StatusCode"], 200)
+        result_payload = json.loads(result["Payload"].read().decode("utf-8"))
+        self.assertIn("message", result_payload)
+        self.assertEqual("success", result_payload["message"])
```

Comment:
Perhaps this could be simplified and improved like this?:
````
expected_payload = {"message": "success"}
with tracer.start_active_span('test'):
    result = self.aws_lambda.invoke(FunctionName=self.function_name, Payload=json.dumps(expected_payload))
self.assertEqual(result["StatusCode"], 200)
result_payload = json.loads(result["Payload"].read().decode("utf-8"))
self.assertDictEqual(expected_payload, result_payload)
````

Reply:
Yeah, I initially began with this approach, but it turns out the payload contains additional headers as well.

```python
{'message': 'success', 'X-INSTANA-L': '1', 'traceparent': '00-00000000000000006b2ac5343fa63053-ce05b7082d194345-01', 'tracestate': 'in=6b2ac5343fa63053;ce05b7082d194345', 'X-INSTANA-T': '6b2ac5343fa63053', 'X-INSTANA-S': 'ce05b7082d194345'}
```